community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/AwaitIndexProcedure.java

```diff
@@ -139,4 +142,13 @@ private String formatIndex( String labelName, String propertyKeyName )
     {
         return format( ":%s(%s)", labelName, propertyKeyName );
     }
+
+    @Override
+    public void close()
+    {
+        if ( statement != null )
+        {
+            statement.close();
```

Comment:
Since you have this if-condition here, did you intend to also set the statement to null after closing so that you would have idempotent close?